lib/model/rwfolder.go

```diff
@@ -87,6 +89,8 @@ type rwFolder struct {
 	copiers      int
 	pullers      int
 	shortID      uint64
+	newVersion   protocol.Vector
```

Comment:
Take a look at the `sharedPullerState` which is passed around to each stage. In there you have `file.Version` (the version of the new file we're about to commit to) and `version` (the version of the old file we had on disk). The naming there isn't super awesome and should probably be clarified, but at least it's commented in https://github.com/syncthing/syncthing/blob/master/lib/model/sharedpullerstate.go#L21